### PR TITLE
Refactor eventsource [Extensions.PersistentStorage]

### PR DESCRIPTION
**Pull request by vishweshbankwar**

### CODE REVIEW

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Extensions.PersistentStorage/FileBlobProvider.cs [hunk 1]
```diff
@@ -161,8 +161,9 @@ private bool CheckStorageSize()
             if (size >= this.maxSizeInBytes)
             {
                 // TODO: check accuracy of size reporting.
-                PersistentStorageEventSource.Log.Warning($"Persistent storage max capacity has been reached. Currently at {size / 1024} KB. " +
-                                                "Please consider increasing the value of storage max size in exporter config.");
+                PersistentStorageEventSource.Log.PersistentStorageWarning(
+                    nameof(FileBlobProvider),
+                    $"Persistent storage max capacity has been reached. Currently at {size / 1024} KB. " + "Please consider increasing the value of storage max size in exporter config.");
```
Comment (TimothyMothra) [inline src/OpenTelemetry.Extensions.PersistentStorage/FileBlobProvider.cs:166]: nit: remove the `+`

```suggestion
                    $"Persistent storage max capacity has been reached. Currently at {size / 1024} KB. Please consider increasing the value of storage max size in exporter config.");
```

File: src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs [hunk 2]
```diff
@@ -16,92 +16,54 @@
...
+            if (this.IsEnabled(EventLevel.Error, EventKeywords.All))
+            {
+                this.PersistentStorageException(className, message, ToInvariantString(ex));
+            }
         }
 
-        [NonEvent]
-        public void Informational(string message, object value = null)
+        [Event(1, Message = "{0}: {1}: {2}", Level = EventLevel.Error)]
+        public void PersistentStorageException(string className, string message, string ex)
```
Comment (reyang) [inline src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs:40]: What is the semantic of `PersistentStorageException`? (e.g. who is the faulty party, who should take action)

File: src/OpenTelemetry.Extensions.PersistentStorage/FileBlob.cs [hunk 3]
```diff
@@ -97,7 +97,7 @@ protected override bool OnTryLease(int leasePeriodMilliseconds)
             }
             catch (Exception ex)
             {
-                PersistentStorageEventSource.Log.Warning($"Acquiring a lease to file {this.FullPath} has failed.", ex);
+                PersistentStorageEventSource.Log.PersistentStorageException(nameof(FileBlob), $"Acquiring a lease to file {this.FullPath} has failed.", ex);
```
Comment (reyang) [inline src/OpenTelemetry.Extensions.PersistentStorage/FileBlob.cs:100]: How frequent would this happen? and how much perf/memory penalty would this introduce?

**Review by reyang — CHANGES_REQUESTED** [review]
Need to clarify the semantics https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/382/files#r884033099.

File: src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs (hunk 2 above)
Comment (reyang) [reply src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs:40]: E.g. who will be consuming these logs, what should they expect to do with these logs, are these logs helpful to drive any meaningful action?

File: src/OpenTelemetry.Extensions.PersistentStorage/FileBlob.cs (hunk 3 above)
Comment (reyang) [reply src/OpenTelemetry.Extensions.PersistentStorage/FileBlob.cs:100]: If this has any chance to be triggered frequently, need to use a heap-alloc-free approach (in other words, no string interpolation).

Comment (vishweshbankwar) [reply src/OpenTelemetry.Extensions.PersistentStorage/FileBlob.cs]: updated - Please review.

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs [hunk 4]
```diff
@@ -16,92 +16,180 @@
...
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
+            {
+                this.CouldNotCreateFileBlob(ToInvariantString(ex));
+            }
+        }
+
+        [NonEvent]
+        public void CouldNotRemoveExpiredBlob(string filePath, Exception ex)
+        {
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
```
Comment (TimothyMothra) [inline src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs:78]: This looks wrong.
Line 78 is checking if `Informational` is enabled.
But the actual event method is `Warning` line 141.

Looks like some of the other `NonEvent` methods have the same mistake.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs:78]: My bad: I will fix it.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs]: fixed

**Review by TimothyMothra — APPROVED** [review]
LGTM

**Pull request merged**